src/lib.rs

```diff
@@ -97,7 +102,7 @@ impl SafetyContext {
 
     /// Gather unconditional and conditional rules to be provided to the seccomp context.
     #[allow(clippy::needless_pass_by_value)]
-    fn gather_rules(rules: impl RuleSet) -> Vec<Rule> {
+    fn gather_rules(rules: impl RuleSet) -> HashMap<syscalls::Sysno, Vec<Rule>> {
         let base_syscalls = rules.simple_rules();
         let mut rules = rules.conditional_rules();
         for syscall in base_syscalls {
```

Comment:
I'm a bit confused by this piece of code; it seems to silently discard simple rules if a conditional rule for that syscall already exists. I'd have expected that to produce an error, since the conditional rule is now preempted.

Reply:
In the libseccomp documentation:
> For example, if a rule is added which allows a given syscall with a specific set of argument values and later a rule is added which allows the same syscall regardless the argument values then the first, more specific rule, is effectively dropped from the filter by the second more generic rule.

However, this behavior seemed strange to me because it seemed like it would be easy to accidentally clobber your own conditional rules.

The compromise I came up with was just dropping the more general rules, so that in the worst case your program doesn't allow something and you have to re-examine your rules, rather than accidentally allowing something you wanted to filter with a condition.